tasks/agent.py

```diff
@@ -661,13 +658,59 @@ def check_supports_python_version(_, check_dir, python):
         for node in ast.walk(tree):
             if isinstance(node, ast.keyword) and node.arg == 'classifiers':
                 classifiers = ast.literal_eval(node.value)
-                print(any(cls.startswith(prefix) for cls in classifiers), end='')
-                return
+                return any(cls.startswith(prefix) for cls in classifiers)
         else:
-            print('False', end='')
+            return False
     else:
-        raise Exit('not a Python project', code=1)
+        return False
+
+
+@task
+def collect_integrations(_, integrations_dir, python_version, target_os, excluded):
+    """
+    Collect and print the list of integrations to install.
+
+    `excluded` is a comma-separated list of directories that don't contain an actual integration
+    """
+    import json
+
+    excluded = excluded.split(',')
+    integrations = []
+
+    for entry in os.listdir(integrations_dir):
+        int_path = os.path.join(integrations_dir, entry)
+        if not os.path.isdir(int_path) or entry in excluded:
+            continue
+
+        manifest_file_path = os.path.join(int_path, "manifest.json")
+
+        # If there is no manifest file, then we should assume the folder does not
+        # contain a working check and move onto the next
+        if not os.path.exists(manifest_file_path):
+            continue
+
+        with open(manifest_file_path) as f:
+            manifest = json.load(f)
+
+        # Figure out whether the integration is supported on the target oS
+        if 'supported_os' in manifest:
+            if target_os not in manifest['supported_os']:
+                continue
+        else:
+            if target_os == 'mac_os':
+                tag = 'Supported OS::macOS'
+            else:
+                tag = f'Supported OS::{target_os.capitalize()}'
+
+        if tag not in manifest['tile']['classifier_tags']:
```

Comment:
If `target_os` is in `manifest['supported_os']` then won't `tag` be undefined here?

Reply:
I was translating the ruby logic here https://github.com/DataDog/datadog-agent/blob/737ed9a3f92baabfaed0fec11f5df1f398b64866/omnibus/config/software/datadog-agent-integrations-py3.rb#L143-L153

IIUC this is because this supports two different versions of the manifest,
[example old](https://github.com/DataDog/integrations-core/blob/4008b99acb629c03c4b7b46aece9f457400cd7e3/glusterfs/manifest.json#L12-L14), [example new](https://github.com/DataDog/integrations-core/blob/master/glusterfs/manifest.json#L15).

Admittedly the `supported_os` check above could probably be removed at this point because I don't think we have any old manifests lying around. But I don't see big harm in keeping it to avoid changing any logic in this PR.

Reply:
Right, but in the previous ruby code, checking for `tag` within the manifest's `classifier_tags` is within the `else` branch where `tag` is always defined. I think your version has moved it up a level of indentation.

Reply:
Let's get rid of the first conditional, which should also address this issue. All manifests should be on v2 now.

Reply:
@ofek OK, thanks for the confirmation that it can be safely removed, done now!

@dustmop Indeed, I see what you mean now, you were absolutely right. It should be fixed now that we agreed to remove the first branch.